modules/videoio/test/test_video_io.cpp

```diff
@@ -616,4 +616,45 @@ static vector<Ext_Fourcc_API> generate_Ext_Fourcc_API_nocrash()
 
 INSTANTIATE_TEST_CASE_P(videoio, Videoio_Writer_bad_fourcc, testing::ValuesIn(generate_Ext_Fourcc_API_nocrash()));
 
+typedef testing::TestWithParam<VideoCaptureAPIs> safe_capture;
+
+TEST_P(safe_capture, draw_even_frames)
+{
+    VideoCaptureAPIs apiPref = GetParam();
+    if (!videoio_registry::hasBackend(apiPref))
+        throw SkipTestException(cv::String("Backend is not available/disabled: ") + cv::videoio_registry::getBackendName(apiPref));
+
+    VideoCapture cap;
+    String video_file = BunnyParameters::getFilename(String(".avi"));
+    EXPECT_NO_THROW(cap.open(video_file, apiPref));
+    if (!cap.isOpened())
+    {
+        std::cout << "SKIP test: backend " << apiPref << " can't open the video: " << video_file << std::endl;
+        return;
+    }
+
+    Mat frames[10];
+    Mat hardCopies[10];
+    for(int i = 0; i < 10; i++)
+    {
+        ASSERT_NO_THROW(cap >> frames[i]);
+        EXPECT_FALSE(frames[i].empty());
+        hardCopies[i] = frames[i].clone();
+    }
+
+    for(int i = 0; i < 10; i+=2)
+        frames[i].setTo(cv::Scalar(255,255,255));
```

Comment:
It is a bad idea to require changing of such buffers.
They can be allocated as read-only (any write attempt will raise a failure).
For example, OpenCL read-only "map" from GPU decoder.

---

Suggestion:
- remove this block
- rename test

Reply:
I renamed the test and removed Mat modification as you suggested, but it's not what OpenCV users expect. Documentation does not provide any cases with read-only Mat object and a lot of examples do camera Mat modification.